vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/FieldBase.java

```diff
@@ -0,0 +1,309 @@
+     */
+    public String getTitle() {
+        return getElement().getProperty("title");
+    }
+
+    /**
+     * The text usually displayed in a tooltip popup when the mouse is over the
+     * field.
+     *
+     * @param title
+     *            the String value to set
+     */
+    public void setTitle(String title) {
+        getElement().setProperty("title", title == null ? "" : title);
+    }
+
+    /**
+     * Specifies if the field value gets automatically selected when the field
+     * gains focus.
+     *
+     * @return <code>true</code> if autoselect is active, <code>false</code>
+     *         otherwise
+     */
+    public boolean isAutoselect() {
+        return getElement().getProperty("autoselect", false);
+    }
+
+    /**
+     * Set to <code>true</code> to always have the field value automatically
+     * selected when the field gains focus, <code>false</code> otherwise.
+     *
+     * @param autoselect
+     *            <code>true</code> to set auto select on, <code>false</code>
+     *            otherwise
+     */
+    public void setAutoselect(boolean autoselect) {
+        getElement().setProperty("autoselect", autoselect);
+    }
+
+    /**
+     * Sets whether the component has an invalid value or not.
+     *
+     * @param invalid
+     *            {@code true} for invalid, {@code false} for valid
+     */
+    public void setInvalid(boolean invalid) {
+        getElement().setProperty("invalid", invalid);
+    }
+
+    /**
+     * Whether the component has an invalid value or not.
+     *
+     * @return the {@code invalid} property from the web component
+     */
+    public boolean isInvalid() {
+        return getElement().getProperty("invalid", false);
+    }
+
+    /**
+     * Specifies that the user must fill in a value.
+     *
+     * @param required
+     *            the boolean value to set
+     */
+    public void setRequired(boolean required) {
+        getElement().setProperty("required", required);
+    }
+
+    /**
+     * Determines whether the field is marked as input required.
+     *
+     * @return {@code true} if the input is required, {@code false} otherwise
+     */
+    public boolean isRequired() {
+        return getElement().getProperty("required", false);
+    }
+
+    /**
+     * Adds the given components as children of this component at the slot
+     * 'prefix'.
+     *
+     * @param components
+     *            The components to add.
+     * @see <a href=
+     *      "https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot">MDN
+     *      page about slots</a>
+     * @see <a href=
+     *      "https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element">Spec
+     *      website about slots</a>
+     */
+    protected void addToPrefix(Component... components) {
+        for (Component component : components) {
+            component.getElement().setAttribute("slot", "prefix");
+            getElement().appendChild(component.getElement());
```

Comment:
Would it make sense to add a universal add to slot method to https://github.com/vaadin/flow-components/blob/master/vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/SlotUtils.java allowing to re-use it here and on other occurrences?

Reply:
Sounds good 👍

Reply:
Another note: should this be moved to the shared package and more used while you guys are at cleaning the components? https://github.com/vaadin/flow-components/blob/master/vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/HasPrefixAndSuffix.java

Reply:
Yes, I was thinking about that too. I will create a PR.

Reply:
Here is a PR that adds `HasPrefix` and `HasSuffix` interfaces: #4458